(GH-2000) Deprecate bolt.yaml (#2397)
puppetlabs/bolt: documentation/configuring_bolt.md

You can view a full list of the available options in [`bolt.yaml`
options](bolt_configuration_reference.md).

#### Updating project-level `bolt.yaml`

> You can update to use the latest Bolt best practices using our migration command:

Reviewer: ```suggestion
You can update your project-level configuration to use `bolt-project.yaml` instead of `bolt.yaml`
with our migration command:
```

_\*nix shell command_

```shell
bolt project migrate
```

_PowerShell cmdlet_

```powershell
Update-BoltProject

> ```

Reviewer: We might want to just create a doc for upgrading to Bolt 3.0 that includes this information, similar to the doc we had for [migrating to Bolt 2.0](https://puppet.com/docs/bolt/latest/migrating_inventory_files.html).

Author: Ok, the doc I added is definitely not comprehensive but I figure we'll build it up as we get closer to 3.0

You can view a full list of the available options in [`bolt.yaml`
options](bolt_configuration_reference.md).

> #### Updating user-level `bolt.yaml`

Reviewer: Ditto above.

To update your user-level `bolt.yaml` to current Bolt practices, you'll want to:

> 1. Move any [transport configuration](bolt_transports_reference.md) to be under the
>    `inventory-config` key, like so:
> 

Reviewer: ```suggestion
1. Move any [transport configuration](bolt_transports_reference.md) options under the
   `inventory-config` key, like so:

```

   ```
   inventory-config:

>        ssh:
>          password: hunter2!
>          user: bolt
>        winrm:
>          password: hunter2!
>          user: bolt

Reviewer: ```suggestion
     ssh:
       password: hunter2!
       user: bolt
     winrm:
       password: hunter2!
       user: bolt
```

You can view a full list of the available options in [`bolt.yaml`
options](bolt_configuration_reference.md).

> #### Updating system-level `bolt.yaml`

Reviewer: Ditto above.

To update your system-level `bolt.yaml` to current Bolt practices, you'll want to:

> 1. Move any [transport configuration](bolt_transports_reference.md) to be under the
>    `inventory-config` key, like so:

Reviewer: ```suggestion
1. Move any [transport configuration](bolt_transports_reference.md) options under the
   `inventory-config` key, like so:
```

   ```
   inventory-config:

>        ssh:
>          password: hunter2!
>          user: bolt
>        winrm:
>          password: hunter2!
>          user: bolt

Reviewer: ```suggestion
     ssh:
       password: hunter2!
       user: bolt
     winrm:
       password: hunter2!
       user: bolt
```
